doc/design/mount-config-design.md

```diff
@@ -0,0 +1,115 @@
+# Automatic filesystem mount configuration design
+
+During startup we need the ability to mount directories from the insecure host into the enclave.
+Currently an application can do this through a system call,
+but this needs to be automated at start-up if the necessary configuration is given.
+Configuration needed for a mount includes the mount source path which is on an insecure filesystem,
+mount target directory within the TEE filesystem,
+and other data relating configuration like if it is read-only or not.
+
```

Comment:
What if the user want to mount to a location from host that already exists in the enclave? For example, the enclave might have files under `/etc`, and if we map host `/etc` to enclave `/etc` with the config file and mount option, what's the behavior? The host /etc overwrites the enclave /etc, or they co-exist?

Reply:
If you mount a source directory over a target directory that has files in then you can no longer see the old files, only the new ones. I believe mount -bind allows you to see what is already there as well.
I assume the same is also the case with a file being mounted into target directory where the file exists... that one will become hidden and the newly mounted one will be seen.
That said, if someone is already connected to a inode (like a bash shell) and is in a directory at the time the mount happens, that directory will still look the same after the mount has happened until the context moves outside the mount location and back in, in which case they will now see the new mounted directory.
Note: this is the linux kernel implementation based on what I read [here](https://unix.stackexchange.com/questions/198542/what-happens-when-you-mount-over-an-existing-folder-with-contents?rq=1)
I do not know the current behaviour in Mystikos